File: leetcode/easy/1313/1313.js
/**
 * Runtime: 100 ms, faster than 50.06%
 * Memory Usage: 42.2 MB, less than 66.48%
 */
var decompressRLElist = function(nums) {
  let result = []

  for (let i=0; i<nums.length; i+=2) {
    let f = nums[i];
    let v = nums[i+1];

    for (let j=0; j<f; ++j) {
      result.push(v);
    }
  }

  return result;
};

File: leetcode/easy/1313/1313.rb
# Runtime: 72 ms, faster than 85.11%
# Memory Usage: 210.5 MB, less than 82.98%
def decompress_rl_elist(nums)
    result = []
    
    for i in (0...nums.size()).step(2) do
        f = nums[i]
        v = nums[i+1]
        
        1.upto(f) do
            result << v 
        end
    end
    
    result
end
